libraries/abstractions/secure_sockets/test/iot_test_tcp.c

```diff
@@ -1554,8 +1555,16 @@ static void prvSOCKETS_NonBlocking_Test( Server_t xConn )
                  */
                 xNumBytesReceived += lNumBytes;
             }
-
-            if( lNumBytes < 0 )
+            
+            /* Since we are in non blocking mode, this error is expected if the TCP stack is
+             * still processing the packet. */
+            if( ( lNumBytes < 0 ) && ( lNumBytes != SOCKETS_EWOULDBLOCK ) )
```

Comment:
not related to your changes but better to have this check as else if so we omit it if we get into the previous one

Reply:
Done.